File: src/UE4_Assets/FCompressedChunk.java
/**
 * 
 */
package UE4_Assets;

import annotation.Int32;
import annotation.Serializable;
import lombok.Data;

/**
 * @author FunGames
 *
 */
@Data
@Serializable
public class FCompressedChunk {
	@Int32 private int uncompressedOffset;
	@Int32 private int uncompressedSize;
	@Int32 private int compressedOffset;
	@Int32 private int compressedSize;
}

File: src/main/kotlin/me/fungames/jfortniteparse/ue4/assets/objects/FCompressedChunk.kt
package me.fungames.jfortniteparse.ue4.assets.objects

import me.fungames.jfortniteparse.ue4.reader.FArchive
import me.fungames.jfortniteparse.ue4.writer.FArchiveWriter

// from Linker.h
class FCompressedChunk {
    var uncompressedOffset: Int
    var uncompressedSize: Int
    var compressedOffset: Int
    var compressedSize: Int

    constructor(Ar: FArchive) {
        uncompressedOffset = Ar.readInt32()
        uncompressedSize = Ar.readInt32()
        compressedOffset = Ar.readInt32()
        compressedSize = Ar.readInt32()
    }

    fun serialize(Ar: FArchiveWriter) {
        Ar.writeInt32(uncompressedOffset)
        Ar.writeInt32(uncompressedSize)
        Ar.writeInt32(compressedOffset)
        Ar.writeInt32(compressedSize)
    }

    constructor(uncompressedOffset: Int, uncompressedSize: Int, compressedOffset: Int, compressedSize: Int) {
        this.uncompressedOffset = uncompressedOffset
        this.uncompressedSize = uncompressedSize
        this.compressedOffset = compressedOffset
        this.compressedSize = compressedSize
    }
}